### PR TITLE
Update DoD

### PR DESCRIPTION
**Pull request by halprin**
## Description

Updated the DoD based on our discussion at the Engineering Block from October 4, 2024.

- Added items for e2e tests.
- Removed PR checks because they will automatically be covered when the `Source code is merged to the main branch` item is checked.

## Issue

_None._

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/story.md
Comment (pluckyswan) [inline .github/ISSUE_TEMPLATE/story.md:47]: Is unit testing now implied/as necessary?

File: .github/ISSUE_TEMPLATE/story.md
Comment (jherrflexion) [inline .github/ISSUE_TEMPLATE/story.md:43]: This is great. Great job.

File: .github/ISSUE_TEMPLATE/story.md
Comment (jherrflexion) [reply .github/ISSUE_TEMPLATE/story.md:47]: I am also curious

Comment (basiliskus) [reply .github/ISSUE_TEMPLATE/story.md:47]: This is enforced by `jacocoTestCoverageVerification` and it's a PR-level requirement. It makes sense to remove it as we wouldn't be able to merge a PR if we had less test coverage. At least in the TI repo

File: .github/ISSUE_TEMPLATE/story.md
Comment (basiliskus) [inline .github/ISSUE_TEMPLATE/story.md:44]: Instead of "additional RS e2e tests", I would just name the assertions. There is only one "test" that loops and applies all the assertions, so it's not expected we'll need more tests. Something like "make sure to add new assertions in the rs-e2e project for any new transformations"

File: .github/ISSUE_TEMPLATE/story.md
Comment (pluckyswan) [reply .github/ISSUE_TEMPLATE/story.md:47]: That's good to know. Looks like SFTP has a threshold.

File: .github/ISSUE_TEMPLATE/story.md
Comment (jorg3lopez) [inline .github/ISSUE_TEMPLATE/story.md:42]: I'm curious to know why this needs to get deleted.

File: .github/ISSUE_TEMPLATE/story.md
Comment (jorg3lopez) [inline .github/ISSUE_TEMPLATE/story.md:43]: Same with this one. Would deleting this entry lead to overlooking potential vulnerabilities or privacy concerns?

File: .github/ISSUE_TEMPLATE/story.md
Comment (halprin) [reply .github/ISSUE_TEMPLATE/story.md]: Did this.

File: .github/ISSUE_TEMPLATE/story.md
Comment (halprin) [reply .github/ISSUE_TEMPLATE/story.md:42]: Removing this doesn't mean we're changing our process.  These were removed because they are automatically accounted for when doing the PR process, and therefore accounted for in the existing `Source code is merged to the main branch` DoD item.

File: .github/ISSUE_TEMPLATE/story.md
Comment (halprin) [reply .github/ISSUE_TEMPLATE/story.md:43]: @jorg3lopez or others, do you think there is a security or privacy gate or other security check that we aren't automatically checking during the PR process that we should consider as part of this DoD item?

Comment (jorg3lopez) [reply .github/ISSUE_TEMPLATE/story.md:43]: I like that we have the process as part of our PR review. For the most part, I think we are doing a great job using workflows and other software to test and detect vulnerabilities. Since we can't merge the PRs if they don't pass those security gates, I think it's ok to delete them here, as long as we keep them as part of the git checks.

**Review by jorg3lopez — APPROVED** [review]
Looks good. The PR review already has steps that cover the deleted tasks from the DoD.

File: .github/ISSUE_TEMPLATE/story.md
Comment (tjohnson7021) [reply .github/ISSUE_TEMPLATE/story.md:42]: Just wondering - I agree @halprin that the PR review process implies being reviewed by a pair, but I'm wondering if the outside reviewer part is implied. I know the PR creator can't approve by function, but git doesn't stop the other collaborator(s) from approving (unless there's something about co-authors Ii don't know).

Comment (halprin) [reply .github/ISSUE_TEMPLATE/story.md:42]: @tjohnson7021, you've got it right.  A collaborator could unilaterally approve it.  And yes, I think it should continue to be ideal that reviewers be outside the people who worked on the PR.  You aren't always guaranteed to be able to get this, but that's OK.

**Pull request merged**
